file/readfile_test.go

```diff
@@ -451,6 +451,61 @@ func Test_getContent(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "file with env var and parse Int",
+			args: args{[]string{"testdata/parseint/file.yaml"}},
+			envVars: map[string]string{
+				"DECK_WRITE_TIMEOUT": "1337",
+			},
+			want: &Content{
+				Services: []FService{
+					{
+						Service: kong.Service{
+							Name:         kong.String("svc1"),
+							Host:         kong.String("mockbin.org"),
+							WriteTimeout: kong.Int(1337),
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "file with env var and parse Int - err on bad value",
+			args: args{[]string{"testdata/parseint/file.yaml"}},
+			envVars: map[string]string{
+				"DECK_WRITE_TIMEOUT": "RIP",
+			},
+			wantErr: true,
+		},
+		{
+			name: "file with env var and parse Float64",
+			args: args{[]string{"testdata/parsefloat/file.yaml"}},
+			envVars: map[string]string{
+				"DECK_FOO_FLOAT": "1337",
+			},
+			want: &Content{
+				Plugins: []FPlugin{
+					{
+						Plugin: kong.Plugin{
+							Name: kong.String("foofloat"),
+							Config: kong.Configuration{
+								"foo": *kong.Float64(1337),
```

Comment:
weirdly the assertions seems to fail on this one if i don't use the value. However it works fine for the `parseInt`.

Reply:
That's because plugins' config stores actual values, not pointers (like for other non-plugin entities). You can rewrite this as `float64(1337)`.